connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/ProtoExpressionSQLBuilder.scala

```diff
@@ -0,0 +1,213 @@
+      case proto.Expression.ExprTypeCase.CAST =>
+        val cast = expr.getCast
+        cast.getCastToTypeCase match {
+          case proto.Expression.Cast.CastToTypeCase.TYPE =>
+            visitCast(expr.getCast)
+          case _ => visitUnexpectedExpr(expr)
+        }
+      case proto.Expression.ExprTypeCase.SORT_ORDER => visitSortOrder(expr.getSortOrder)
+      case _ =>
+        throw InvalidPlanInput(
+          s"Expression with ID: ${expr.getExprTypeCase.getNumber} is not supported")
+    }
+  }
+
+  private def visitLiteral(literal: proto.Expression.Literal): String = {
+    val value = LiteralValueProtoConverter.toCatalystValue(literal)
+    literal.getLiteralTypeCase match {
+      case proto.Expression.Literal.LiteralTypeCase.STRING =>
+        s"'$value'"
+      case _ =>
+        s"$value"
+    }
+  }
+
+  private def visitUnresolvedAttribute(attr: proto.Expression.UnresolvedAttribute): String =
+    attr.getUnparsedIdentifier
+
+  private def visitUnresolvedFunction(fun: proto.Expression.UnresolvedFunction): String = {
+    val functionName = fun.getFunctionName
+    functionName match {
+      case "=" | "<=>" | "<" | "<=" | ">" | ">=" =>
+        visitBinaryComparison(
+          functionName,
+          generate(fun.getArguments(0)),
+          generate(fun.getArguments(1)))
+      case "+" | "-" | "*" | "/" | "%" | "&" | "|" | "^" =>
+        visitBinaryArithmetic(
+          functionName,
+          generate(fun.getArguments(0)),
+          generate(fun.getArguments(1)))
+      case "and" | "or" =>
+        visitOr(functionName, generate(fun.getArguments(0)), generate(fun.getArguments(1)))
+      case "negative" =>
+        visitUnaryArithmetic("-", generate(fun.getArguments(0)))
+      case "!" =>
+        visitUnaryArithmetic("NOT", generate(fun.getArguments(0)))
+      case "when" =>
+        visitCaseWhen(fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      case "isNull" =>
+        visitIsNull(generate(fun.getArguments(0)))
+      case "isNotNull" =>
+        visitIsNotNull(generate(fun.getArguments(0)))
+      case "in" =>
+        visitJoin(fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      case "like" =>
+        visitLike(generate(fun.getArguments(0)), generate(fun.getArguments(1)))
+      case "isNaN" =>
+        visitCommonFunction("isnan", fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      case "rlike" =>
+        visitCommonFunction("RLIKE", fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      case "substr" =>
+        visitCommonFunction("substring", fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      case "ilike" | "contains" | "startswith" | "endswith" =>
+        visitCommonFunction(functionName, fun.getArgumentsList.asScala.toSeq.map(generate(_)))
+      // TODO: Support more functions
+    }
+  }
+
+  private def visitBinaryComparison(name: String, l: String, r: String): String =
+    s"($l $name $r)"
```

Comment:
I think this is too much for cosmetic changes, and even with this it can't address all the cases, right? e.g., unresolved column such as `a AS b`. In non Spark Connect, it will be shown `b` IIRC.

Reply:
Yeah. we can add each case on the way. This implementation based on the discussion https://github.com/apache/spark/pull/40467#discussion_r1144196759

Reply:
The current implementation doesn't guarantee the coverage for all the cases. But we can add them in following PRs.